test/e2e/resize_nodes.go

```diff
@@ -421,6 +421,14 @@ var _ = Describe("Nodes", func() {
 			if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes, 10*time.Minute); err != nil {
 				Failf("Couldn't restore the original cluster size: %v", err)
 			}
+			// Many e2e tests assume that the cluster is fully healthy before they start.  Wait until
+			// the cluster is restored to health
+			By("waiting for system pods to successfully restart")
+			pods, err := framework.Client.Pods("kube-system").List(labels.Everything(), fields.Everything())
```

Comment:
What happens here is that the pods are already created and running on node N, then we resize downward, and node N is destroyed, so the pods move to node M, so the pods exist, but they're not scheduled.  So we need to wait for them to start running.